src/sync/sync_permission.cpp

```diff
@@ -170,16 +171,22 @@ void Permissions::set_permission(std::shared_ptr<SyncUser> user,
                                  PermissionChangeCallback callback,
                                  const ConfigMaker& make_config)
 {
+    using namespace std::chrono;
     const auto realm_url = user->server_url() + permission.path;
     auto realm = Permissions::management_realm(std::move(user), make_config);
     CppContext context;
 
+    // Get the current time.
+    int64_t ns_since_epoch = duration_cast<nanoseconds>(system_clock::now().time_since_epoch()).count();
```

Comment:
Since what we care about is the time since the UNIX epoch, I'd think we could populate a `struct tm` with the UNIX epoch, convert that to a `time_t`, then use `std::chrono::system_clock::from_time_t` to convert that to a `time_point` that we can then subtract from the current time point.